### PR TITLE
Fix bug when doing linked schema change.

### PR DESCRIPTION
**Pull request by chaoyli**
1. rollup will create table with num_key_columns is less than base table.
2. adding column to existed table, num_key_columns is larger than base table.
When meets two events, we should handle them carefully.

### CODE REVIEW

File: be/src/olap/rowset/alpha_rowset_writer.cpp [hunk 1]
```diff
@@ -129,7 +129,28 @@ OLAPStatus AlphaRowsetWriter::add_rowset(RowsetSharedPtr rowset) {
...
+
+OLAPStatus AlphaRowsetWriter::add_rowset_for_linked_schema_change(
+        RowsetSharedPtr rowset, const SchemaMapping& schema_mapping) {
+    _need_column_data_writer = false;
+    // this api is for LinkedSchemaChange
+    // use create hard link to copy rowset for performance
+    // this is feasible because LinkedSchemaChange is done on the same disk
+    AlphaRowsetSharedPtr alpha_rowset = std::dynamic_pointer_cast<AlphaRowset>(rowset);
+    for (auto& segment_group : alpha_rowset->_segment_groups) {
+        RETURN_NOT_OK(_init());
```
Comment (imay) [inline be/src/olap/rowset/alpha_rowset_writer.cpp]: _init will be called many times

File: be/src/olap/rowset/segment_group.cpp [hunk 2]
```diff
@@ -234,27 +235,42 @@ bool SegmentGroup::delete_all_files() {
...
+
         WrapperField* first = WrapperField::create(column);
         DCHECK(first != NULL) << "failed to allocate memory for field: " << i;
-        first->copy(zone_map_fields[i].first);
 
         WrapperField* second = WrapperField::create(column);
         DCHECK(second != NULL) << "failed to allocate memory for field: " << i;
-        second->copy(zone_map_fields[i].second);
+
+        //for new key column, use default value to fill into column_statistics
```
Comment (imay) [inline be/src/olap/rowset/segment_group.cpp]:
```suggestion
        // for new key column, use default value to fill into column_statistics
```

File: be/src/olap/rowset/segment_group.cpp [hunk 3]
```diff
@@ -234,27 +235,42 @@ bool SegmentGroup::delete_all_files() {
...
+        const SchemaMapping& schema_mapping) {
     //When add rollup tablet, the base tablet index maybe empty
     if (zone_map_fields.size() == 0) {
         return OLAP_SUCCESS;
     }
 
-    //Should use _num_key_columns, not zone_map_fields.size()
-    //as rollup tablet num_key_columns will less than base tablet zone_map_fields.size().
-    //For LinkedSchemaChange, the rollup tablet keys order is the same as base tablet
+    //1. rollup tablet num_key_columns() will less than base tablet zone_map_fields.size().
```
Comment (imay) [inline be/src/olap/rowset/segment_group.cpp]:
```suggestion
    // 1. rollup tablet num_key_columns() will less than base tablet zone_map_fields.size().
```

File: be/src/olap/rowset/segment_group.cpp [hunk 4]
```diff
@@ -234,27 +235,42 @@ bool SegmentGroup::delete_all_files() {
...
+    //   For LinkedSchemaChange, the rollup tablet keys order is the same as base tablet
+    //2. adding column to existed table, num_key_columns() will larger than
```
Comment (imay) [inline be/src/olap/rowset/segment_group.cpp]:
```suggestion
    // 2. adding column to existed table, num_key_columns() will larger than
```

**Review by imay — CHANGES_REQUESTED** [review]

**Review by imay — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**